### DISCUSSION TITLE
Unning GitHub Action setup-node in sub directory

### DISCUSSION THREAD

**Discussion by nicod-pc** [community/community#25647] [Actions]
<p>Hi, I have a repository with a npm project in a sub directory. I liked to use <code>actions/setup-node@v2</code> as described in <a href="https://docs.github.com/en/actions/guides/publishing-nodejs-packages" class="inline-onebox" rel="noopener nofollow ugc">Publishing Node.js packages - GitHub Docs</a>. But when using <code>working-directory: dirname</code> it says: <code>The workflow is not valid. .github/workflows/experiment.yml (Line: 15, Col: 9): Unexpected value 'working-directory'</code> but there isn’t a propery documented for <code>actions/setup-node@v2</code> where I can define the directory. Does someone has any ideas?</p>
<p>Git Structure</p>
<ul>
<li>/
<ul>
<li>/.github/workflow/experiment.yml</li>
<li>/dirname
<ul>
<li>/dirname/package.json</li>
<li>/dirname/index.js</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><code>/.github/workflow/experiment.yml</code>:</p>
<pre><code class="lang-auto">name: Node.js Package
on:
  push:
    branches: [main]
jobs:
  build:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      packages: write
    steps:
      - uses: actions/checkout@v2
      # Setup .npmrc file to publish to GitHub Packages
      - uses: actions/setup-node@v2
        working-directory: dirname
        with:
          node-version: '12.x'
          registry-url: 'https://npm.pkg.github.com'
          # Defaults to the user or organization that owns the workflow file
          scope: '@company'
      - run: npm install
        working-directory: dirname
      - run: npm publish
        working-directory: dirname
        env:
          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>Does someone has an idea? It doesn’t feels like an unusual requirement to have a sub directory with a node project. So I expect it should be working somehow.</p>

**Comment by mickeygousset** [comment]
<p>the working-directory keyword only works with a run command.  The setup-node action isn’t designed to use that keyword.  Why do you have to target that subfolder? WHy not just use setup-node to setup the version on the runner to be used? Then it will be used by your following run steps.</p>

**Comment by nicod-pc** [comment]
<p>So the action should created the .npmrc with the given <code>registry-url</code> and <code>scope</code> is in the correct location, not on the root but on the subfolder.</p>

**Comment by naomiaro** [comment]
<p>You can set <code>cache-dependency-path</code>  <a href="https://naomiaro.hashnode.dev/nodejs-ci-github-action" class="inline-onebox" rel="noopener nofollow ugc">Node.js CI Github Action Monorepo modifications</a></p>

**Reply by sombriks** [reply]
thanks for the link! saved my day!

**Reply by Karamellwuerfel** [reply]
This!!! Thank you so much.

**Comment by nicod-pc** [comment, marked as answer]
<p>Thanks. It’s not about caching. It’s to set registry url for publication. I’ve solved it now with.</p>
<pre><code class="lang-auto">    - name: Configure NPM
      working-directory: dirname
      run: |
        echo "//npm.pkg.github.com/:_authToken=\${NODE_AUTH_TOKEN}" &gt;&gt; .npmrc
        echo "always-auth: true" &gt;&gt; .npmrc
</code></pre>
